pkg/gallery/delete.go

```diff
@@ -11,6 +11,18 @@ import (
 func (s *Service) Destroy(ctx context.Context, i *models.Gallery, fileDeleter *image.FileDeleter, deleteGenerated, deleteFile bool) ([]*models.Image, error) {
 	var imgsDestroyed []*models.Image
 
+	mqb := s.ChapterRepository
+	chapters, err := mqb.FindByGalleryID(ctx, i.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, m := range chapters {
+		if err := DestroyChapter(ctx, m, mqb); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
This should be unnecessary if the relationships cascade correctly.

Reply:
This one is... unfortunately necessary. Deletion does not work with an existing chapter if this code is not present. The Error is:
```destroying galleries: executing `DELETE FROM `galleries` WHERE (`galleries`.`id` IN (MYGALLERYID))` [[]]: FOREIGN KEY constraint failed```
I am not sure how this happens. I can imagine what you mean with cascading, but I don't know how it is done in this body of code. If it should be implemented here, something is wrong probably. I need your assistance to find the location where this is realized.
This is equivalent to how it is done for markers when deleting scenes, see:
`pkg/scene/delete.go`  Line 131 following.